apps/prairielearn/src/pages/instructorAssessmentQuestions/components/EditQuestionModal.tsx

```diff
@@ -0,0 +1,491 @@
+import { Modal } from 'react-bootstrap';
+
+import { useEffect, useMemo, useState } from '@prairielearn/preact-cjs/hooks';
+
+import type { QuestionAlternativeJson, ZoneQuestionJson } from '../../../schemas/infoAssessment.js';
+
+export function EditQuestionModal({
+  question,
+  alternativeGroup,
+  showEditModal,
+  onHide,
+  handleUpdateQuestion,
+  assessmentType,
+  addQuestion,
+  qidValidationError,
+}: {
+  question: ZoneQuestionJson | QuestionAlternativeJson;
+  alternativeGroup?: ZoneQuestionJson | null;
+  showEditModal: boolean;
+  onHide: () => void;
+  handleUpdateQuestion: (updatedQuestion: any) => void;
+  assessmentType: 'Homework' | 'Exam';
+  addQuestion?: boolean;
+  qidValidationError?: string;
+}) {
+  // Determine if we're editing an alternative (has a parent group)
+  const isAlternative = !!alternativeGroup;
+
+  // Determine which property was originally set for points (points vs autoPoints)
+  // Check own values first, then inherited values
+  const originalPointsProperty = useMemo<'points' | 'autoPoints'>(() => {
+    // Check own question first
+    if (question.points !== undefined) return 'points';
+    if (question.autoPoints !== undefined) return 'autoPoints';
+    // Check inherited from alternative group
+    if (isAlternative) {
+      if (alternativeGroup.points !== undefined) return 'points';
+      if (alternativeGroup.autoPoints !== undefined) return 'autoPoints';
+    }
+    // Default to 'autoPoints' for homework, 'points' for exams
+    return assessmentType === 'Exam' ? 'points' : 'autoPoints';
+  }, [question, alternativeGroup, isAlternative, assessmentType]);
+
+  // Determine which max property was originally set for auto grading (maxPoints vs maxAutoPoints)
+  // Prefer matching the points property (points → maxPoints, autoPoints → maxAutoPoints)
```

Comment:
I don't think this should be a "prefer", it must match to sync.

https://github.com/PrairieLearn/PrairieLearn/blob/37983bfe25e44d328e37c1c524afef4f7f03c995/apps/prairielearn/src/sync/course-db.ts#L1287-L1296